test/gtest/mha_find20.cpp

```diff
@@ -0,0 +1,395 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "test.hpp"
+#include "get_handle.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include "../workspace.hpp"
+
+#include <miopen/mha/mha_descriptor.hpp>
+#include <miopen/mha/solvers.hpp>
+#include <miopen/mha/invoke_params.hpp>
+
+#include <miopen/miopen.h>
+
+#include <miopen/solution.hpp>
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+using namespace miopen;
+
+struct TensorStruct
+{
+    TensorStruct(bool isFloat = true) : isFloatTensor(isFloat) {}
+
+    bool isFloatTensor;
+    tensor<float> tensorFloat;
+
+    Allocator::ManageDataPtr gpuBuffer;
+
+    // TODO Unsued for now
+    // tensor<unit64_t> tensorUint64;
+};
+
+typedef std::shared_ptr<TensorStruct> TensorStructPtr;
+
+typedef std::map<miopenTensorArgumentId_t, TensorStructPtr> TensorStructMap;
+
+class MhaFind20Test
+{
+public:
+    MhaFind20Test(bool forward) : problem(nullptr), isForward(forward) { Initialize(); }
+
+    std::vector<miopenSolution_t> TestFindSolutions(Handle& handle)
+    {
+        std::cerr << "Testing miopenFindSolutions..." << std::endl;
+
+        auto solutions = std::vector<miopenSolution_t>{};
+        std::size_t found;
+
+        solutions.resize(16);
+
+        EXPECT_EQUAL(miopenFindSolutions(
+                         &handle, problem, nullptr, solutions.data(), &found, solutions.size()),
+                     miopenStatusSuccess);
+        EXPECT_TRUE(found > 0);
+
+        solutions.resize(found);
+
+        std::cerr << "Finished testing miopenFindSolutions." << std::endl;
+        return solutions;
+    }
+
+    void TestSolutionAttributes(const std::vector<miopenSolution_t>& solutions)
+    {
+        std::cerr << "Testing miopenGetSolution<Attribute>..." << std::endl;
+
+        for(const auto& solution : solutions)
+        {
+            float time;
+            std::size_t workspace_size;
+            uint64_t solver_id;
+
+            EXPECT_EQUAL(miopenGetSolutionTime(solution, &time), miopenStatusSuccess);
+            EXPECT_EQUAL(miopenGetSolutionWorkspaceSize(solution, &workspace_size),
+                         miopenStatusSuccess);
+            EXPECT_EQUAL(miopenGetSolutionSolverId(solution, &solver_id), miopenStatusSuccess);
+        }
+
+        std::cerr << "Finished testing miopenGetSolution<Attribute>." << std::endl;
+    }
+
+    void TestRunSolutionsForward(Handle& handle, const std::vector<miopenSolution_t>& solutions)
+    {
+        std::cerr << "Testing a forward solution..." << std::endl;
+
+        const unsigned int numTensors = tensors.size();
+
+        auto arguments = std::make_unique<miopenTensorArgument_t[]>(numTensors);
+
+        std::vector<miopenTensorDescriptor_t> descVector(numTensors);
+
+        int i = 0;
+        for(const auto& it : tensors)
```

Comment:
It looks a bit clearer.
```suggestion
        for(const auto& [id, tensor] : tensors)
```

Reply:
For me such a syntax construction looks really heavy for its really simple purpose its intended for. So, would prefer my option